classes/03class/exercises/c03-aws03/alanlima/README.md

```diff
@@ -0,0 +1,135 @@
+    
+    ```bash
+    # Modify VPC to enable DNS support
+    $ aws ec2 modify-vpc-attribute --vpc-id vpc-027019faaf6756407 --enable-dns-hostnames "{\"Value\":true}"
+    $ aws ec2 modify-vpc-attribute --vpc-id vpc-027019faaf6756407 --enable-dns-support "{\"Value\":true}"
+    ```
+
+- Command used to add any new records to the zone
+
+    - *Sub Domain*: **c03aws03.devops.rocks**
+
+    ```bash
+    $ aws route53 change-resource-record-sets \
+        --hosted-zone-id Z06064411OE8IA6HY6Q6G \
+        --change-batch file://private-route-record-set.json
+    ```
+        
+    ```yaml
+    ChangeInfo:
+      Comment: Add CNAME to Load Balancer
+      Id: /change/C06262922CL2RPILV3PAJ
+      Status: PENDING
+      SubmittedAt: '2020-07-09T09:17:10.505000+00:00'
+    ```
+    
+    **[private-route-record-set.json](private-route-record-set.json)**
+    
+    ```json
+    {
+        "Comment": "Add CNAME to Load Balancer",
+        "Changes": [
+            {
+                "Action": "UPSERT",
+                "ResourceRecordSet": {
+                    "Name": "c03aws03.devops.rocks",
+                    "Type": "CNAME",
+                    "TTL": 600,
+                    "ResourceRecords": [
+                        {
+                            "Value": "devopsacademy-lb-1691285263.ap-southeast-2.elb.amazonaws.com"
+                        }
+                    ]
+                }
+            }
+        ]
+    }
+    ```
+    
+    * Using UPSERT because it is practical and I may need to update it anyway
+
+- Checking that it actually works..
+
+    ```bash
+    $ ssh ec2-user@13.236.232.76
+    Last login: Thu Jul  9 08:57:30 2020 from 122-148-135-233.sta.wbroadband.net.au
+    
+           __|  __|_  )
+           _|  (     /   Amazon Linux 2 AMI
+          ___|\___|___|
+    
+    https://aws.amazon.com/amazon-linux-2/
+    [ec2-user@ip-172-16-10-104 ~]$ curl c03aws03.devops.rocks
+    ip-172-16-10-117.ap-southeast-2.compute.internal
+    
+    [ec2-user@ip-172-16-10-104 ~]$ curl c03aws03.devops.rocks
+    ip-172-16-10-117.ap-southeast-2.compute.internal
+    
+    [ec2-user@ip-172-16-10-104 ~]$ curl c03aws03.devops.rocks
+    ip-172-16-10-104.ap-southeast-2.compute.internal
+    
+    [ec2-user@ip-172-16-10-104 ~]$ curl c03aws03.devops.rocks
+    ip-172-16-10-104.ap-southeast-2.compute.internal
+    
+    [ec2-user@ip-172-16-10-104 ~]$ curl c03aws03.devops.rocks
+    ip-172-16-10-117.ap-southeast-2.compute.internal
+    
+    [ec2-user@ip-172-16-10-104 ~]$ curl c03aws03.devops.rocks
+    ip-172-16-10-117.ap-southeast-2.compute.internal
+    ```
+
+- Post any findings, how option you decided to go with, records created, domain created, and anything else that you find interesting.
+
+    - I was going to do the public hosted zone, but it took a while until I receive the e-mail for the free domain registration. Then, I started the private, make it works, got excited, received a verification e-mail, did the public zone as well 😅
```

Comment:
Nice! Yes, it takes some time to get the email and then several hours for the DNS name to propagate for the public one.

Reply:
Actually, the DNS propagation was fast. It took only a few seconds between setting the namespaces and be able to use the domain.